Server/App/api/discover.py

```diff
@@ -42,15 +42,12 @@ def get(self, page, connector):
 
 class SearchHandler(BaseHandler):
     @authorize_request_and_create_db_connector
-    @require_query_string_params('query', 'tags', 'page')
-    def get(self, query, tags_json, page, connector):
+    @require_query_string_params('query')
+    @optional_query_string_params('page')
+    def get(self, page, query, connector):
         """Retrieve metadata for decks matching the search string."""
 
-        tags = json.loads(tags_json)
-
-        tags_string = ",".join(tags)
-
-        decks = connector.call_procedure('SEARCH_DECKS', query, tags_string, len(tags), page)
+        decks = connector.call_procedure('SEARCH_DECKS', query, page)
```

Comment:
For all of these `page` optional params, we're going to have to do a `None` check. If it's `None`, we're going to have to set it to 0.